tensornetwork/network.py

```diff
@@ -713,13 +717,40 @@ def contract_between(
         axes1.append(edge.axis2)
         axes2.append(edge.axis1)
 
+    if output_edge_order:
+      # Determine heuristically if output transposition can be minimized by 
+      # flipping the arguments to tensordot.
+      node1_output_axes = []
+      node2_output_axes = []
+      for (i, edge) in enumerate(output_edge_order):
+        if edge in shared_edges:
+          raise ValueError(
+            "Edge '{}' in output_edge_order is shared by the nodes to be "
+            "contracted: '{}' and '{}'.".format(edge, node1, node2)
+          )
+        edge_nodes = set(edge.get_nodes())
+        if node1 in edge_nodes:
+          node1_output_axes.append(i)
+        elif node2 in edge_nodes:
+          node2_output_axes.append(i)
+        else:
+          raise ValueError(
+            "Edge '{}' in output_edge_order is not connected to node '{}' or "
```

Comment:
Add test for this case too. Generally, strive to have 100% test converge when ever possible. It makes refactoring and fixing bugs waaaaayyyyy easier in the long run.

Reply:
This is also covered by the test I added.